File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        stat = {}
        for num in nums:
            if num not in stat:
                stat[num] = 0
            stat[num] += 1
        res = [[] for _ in range(len(nums) + 1)]
        for key in stat:
            res[stat[key]].append(key)
        ans = []
        print(res)
        for item in reversed(res):
            if len(item) == 0:
                continue
            ans.extend(item)
            if len(ans) >= k:
                break
        return ans
            

File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int,int> stat;
        for (const auto& i : nums){
            stat[i]++;
            if(stat[i] > 1) return true;
        }
        return false;
    }
};
